core/src/main/java/com/amplifyframework/analytics/UserProfile.java

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.analytics;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.Objects;
+
+/**
+ * Represents user specific data such as name, email, plan, location etc.
+ */
+public final class UserProfile {
+    private final String name;
+    private final String email;
+    private final String plan;
+    private final Location location;
+    private final Properties customProperties;
+
+    public UserProfile(@NonNull Builder builder) {
```

Comment:
If you make this as `private UserProfile(Builder builder)`, then only you will be able to call this constructor, and you don't have to worry about the builder being null (you'll be able to ensure that, with your own code in this file.)